.claude/scripts/git-safety-contract.test.sh

```
@@ -0,0 +1,140 @@
+#     its ignored-file protection. Without this assertion a later editor closes 2d's gap by adding the
+#     flag — reintroducing every one of those, with a rule that looks more complete than before.
+assert_section '2833' \
+  "the Git safety section no longer points at the follow-up issue for safe submodule detaching — without it the stated hazard reads as unsolved-and-unowned, and the next editor is likely to 'fix' it with \`--recurse-submodules\`, which is measurably unsafe in this repo's linked-worktree flow"
+
+# 2b. THE PR-HEAD ASSOCIATION. \`<sha>\` is deliberately the general placeholder — it is this
+#     contract's convention (27 backticked commands use it, against 1 for \`<headRefOid>\`), and the
+#     rule covers any commit, not only a PR head. But the motivating case IS the PR head, so the
+#     value's identity must be stated or the prescription is unactionable exactly where it is most
+#     needed. Naming \`headRefOid\` also ties this to the commit *Merge policy* pins, which is what
+#     makes "the worktree you evaluated" and "the commit you merged" the same claim.
+assert_section '**`headRefOid`**' \
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)